clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -529,6 +581,14 @@ public KafkaChannel channel(String id) {
     }
 
     /**
+     * Return the channel with the specified id if it was disconnected, but not yet closed
+     * since there are outstanding messages to be processed.
+     */
+    public KafkaChannel closingChannel(String id) {
```

Comment:
Since this is a public api that any caller could use, it seems that it should be in the Selectable interface since we may need to mock it during testing?